zerver/views/home.py

```diff
@@ -157,6 +158,22 @@ def home_real(request: HttpRequest) -> HttpResponse:
         user_profile = None
         realm = get_valid_realm_from_request(request)
 
+        # TODO: Allow user to directly login as web_public_visitor
+        # if `next` contains a web-public-stream.
+
+        # User clicked "Anonymous login" on HOME_NOT_LOGGED_IN page.
+        if request.POST.get('prefers_web_public_view') == 'true':
+            request.session['prefers_web_public_view'] = True
+            # We redirect here to avoid "Confirm form resubmission"
+            # prompt by a modern browser.
+            redirect_to = get_safe_redirect_to(request.POST.get('next'), realm.uri)
+            return HttpResponseRedirect(redirect_to)
+
+        prefers_web_public_view = request.session.get('prefers_web_public_view')
+        # For users who haven't clicked "Anonymous login" we redirect them to login page.
+        if not prefers_web_public_view:
+            return redirect_to_login(next='/')
```

Comment:
@timabbott is this the correct place for this logic? It feels like this doesn't belong here.